dsb-integration/src/test/java/com.spectralogic.dsbrowser.integration/tasks/CreateFolderTaskTest.java

```diff
@@ -0,0 +1,137 @@
+        new JFXPanel();
+        Platform.runLater(() -> {
+            final SavedSession savedSession = new SavedSession(
+                    TEST_ENV_NAME,
+                    client.getConnectionDetails().getEndpoint(),
+                    "80",
+                    null,
+                    new SavedCredentials(
+                            client.getConnectionDetails().getCredentials().getClientId(),
+                            client.getConnectionDetails().getCredentials().getKey()),
+                    false,
+                    false);
+            session = new CreateConnectionTask().createConnection(SessionModelService.setSessionModel(savedSession, false), resourceBundle, buildInfoService);
+            try {
+                envDataPolicyId = IntegrationHelpers.setupDataPolicy(TEST_ENV_NAME, false, ChecksumType.Type.MD5, client);
+                envStorageIds = IntegrationHelpers.setup(TEST_ENV_NAME, envDataPolicyId, client);
+                HELPERS.ensureBucketExists(CREATE_FOLDER_TASK_TEST_BUCKET_NAME, envDataPolicyId);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @AfterClass
+    public static void teardown() throws IOException {
+        IntegrationHelpers.teardown(TEST_ENV_NAME, envStorageIds, client);
+        client.close();
+    }
+
+    @Test
+    public void createFolder() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final String folderName = "testFolder_" + LocalDateTime.now().getSecond();
+        Platform.runLater(() -> {
+            try {
+
+                final CreateFolderModel createFolderModel = new CreateFolderModel(session.getClient(), CREATE_FOLDER_TASK_TEST_BUCKET_NAME, CREATE_FOLDER_TASK_TEST_BUCKET_NAME);
+                final String location = PathUtil.getFolderLocation(createFolderModel.getLocation(),
+                        createFolderModel.getBucketName());
+
+                //Instantiating create folder task
+                final CreateFolderTask createFolderTask = new CreateFolderTask(session.getClient(),
+                        createFolderModel.getBucketName(),
+                        folderName,
+                        PathUtil.getDs3ObjectList(location, folderName),
+                        null,
+                        null);
+                workers.execute(createFolderTask);
+
+                //Validating test case
+                createFolderTask.setOnSucceeded(event -> {
+                    successFlag = true;
+                    //Releasing main thread
+                    latch.countDown();
+                });
+                createFolderTask.setOnFailed(event -> {
+                    latch.countDown();
+                });
+                createFolderTask.setOnCancelled(event -> {
+                    latch.countDown();
+                });
+            } catch (final Exception e) {
+                e.printStackTrace();
```

Comment:
If this is a legitimate error condition, then let it fall out to the junit runner instead of handling it here, and swallowing the exception.